de.dlr.sc.virsat.fdir.core/src/de/dlr/sc/virsat/fdir/core/markov/modelchecker/MarkovModelChecker.java

```diff
@@ -154,19 +159,18 @@ public ModelCheckingStatistics getStatistics() {
 
 	@Override
 	public void visit(Reliability reliabilityMetric, SubMonitor subMonitor) {
+		probabilityDistribution = getInitialProbabilityDistribution();
+
 		if (tmTerminal == null) {
-			Matrix matrix = new Matrix(mc);
-			tmTerminal = matrix.createTransitionMatrix(true, delta);
+			matrixFactory.setMc(mc);

```

Comment:
I think it would be simpler to pass this in the getTransitionMatrix(...) also to rename that method back to createTransitionMatrix(...)